component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/chain/ChainedInput.java

```diff
@@ -88,7 +93,15 @@ public void stop() {
         }
     }
 
-    private void initDelegate() {
-        delegate = parent.getIterator().hasNext() ? parent.getIterator().next().create() : null;
+    private Input initDelegate() {
```

Comment:
with this return type it's more: "pickDelegate"
because you don't set a field value here anymore and nothing is initialized inside the method
we can say that it "startDelegate" cause it's executed the method